test/bats.bats

```diff
@@ -1001,3 +1000,87 @@ EOF
   run stars
   [ "${lines[0]}" = '**' ]
 }
+
+
+@test "--trace: 0 + BATS_TRACE_LEVEL=0 doesn't show anything on failure" {
+  OUTPUT="$(bats --trace 0 --filter "test 1" "$FIXTURE_ROOT/read_from_stdin.bats")"
+  [[ "${OUTPUT[@]}" == "1..1
+ok 1 test 1" ]]
+  BATS_TRACE_LEVEL=0
+  OUTPUT="$(bats --filter "test 1" "$FIXTURE_ROOT/read_from_stdin.bats")"
+  [[ "${OUTPUT[@]}" == "1..1
+ok 1 test 1" ]]
+  unset BATS_TRACE_LEVEL
```

Comment:
I believe that is another problem of having the internal var being the same name as the external one.